ballerina/xml_api.bal

```diff
@@ -0,0 +1,365 @@
+# ```ballerina
+# json data = {
+#     name: "John",
+#     age: 30
+# };
+# xml? xmlValue = check xmldata:fromJson(data);
+# ```
+#
+# + jsonValue - The JSON source to be converted to XML
+# + options - The `xmldata:JsonOptions` record for JSON to XML conversion properties
+# + return - XML representation of the given JSON if the JSON is
+# successfully converted or else an `xmldata:Error`. The `()` value is not returned
+public isolated function fromJson(json jsonValue, JsonOptions options = {}) returns xml?|Error {
+    string? rootTag = options.rootTag;
+    map<string> allNamespaces = {};
+    if !isSingleNode(jsonValue) {
+        addNamespaces(allNamespaces, check getNamespacesMap(jsonValue, options, {}));
+        return getElement(rootTag is string ? rootTag : "root",
+                          check traverseNode(jsonValue, allNamespaces, {}, options), allNamespaces, options,
+                          check getAttributesMap(jsonValue, options, allNamespaces));
+    } else {
+        map<json>|error jMap = jsonValue.ensureType();
+        if jMap is map<json> {
+            if jMap.length() == 0 {
+                return xml ``;
+            }
+            json value = jMap.toArray()[0];
+            addNamespaces(allNamespaces, check getNamespacesMap(value, options, {}));
+            if value is json[] {
+                return getElement(rootTag is string ? rootTag : "root",
+                                  check traverseNode(value, allNamespaces, {}, options, jMap.keys()[0]),
+                                  allNamespaces, options, check getAttributesMap(value, options, allNamespaces));
+            } else {
+                string key = jMap.keys()[0];
+                if key == CONTENT {
+                    return xml:createText(value.toString());
+                }
+                xml output = check getElement(jMap.keys()[0], check traverseNode(value, allNamespaces, {}, options),
+                                            allNamespaces, options,
+                                            check getAttributesMap(value, options, allNamespaces));
+                if rootTag is string {
+                    return xml:createElement(rootTag.toString(), {}, output);
+                }
+                return output;
+            }
+        }
+        if jsonValue !is null {
+            return xml:createText(jsonValue.toString());
+        } else {
+            return xml ``;
+        }
+    }
+}
+
+isolated function traverseNode(json jNode, map<string> allNamespaces, map<string> parentNamespaces, JsonOptions options,
+                                string? key = ()) returns xml|Error {
+    map<string> namespacesOfElem = {};
+    string attributePrefix = options.attributePrefix;
+    xml xNode = xml ``;
+    if jNode is map<json> {
+        foreach [string, json] [k, value] in jNode.entries() {
+            string jsonKey = k.trim();
+            if !jsonKey.startsWith(attributePrefix) {
```

Comment:
Check the inverse and continue/return early as applicable to reduce nesting. Check wherever applicable.